src/chunk.c

```diff
@@ -1051,6 +1186,172 @@ chunk_find_all_oids(Hyperspace *hs, List *dimension_vecs, LOCKMODE lockmode)
 	return oid_list;
 }
 
+/* Implementation of show_chunks that also take the caller name
+ * as argument so error messages refer to the function
+ * that the user actually called. This is necessary because
+ * show_chunk_impl is used by drop_chunks and export_chunks
+ */
+Datum
+ts_chunk_show_chunks(PG_FUNCTION_ARGS)
+{
+	FuncCallContext *funcctx;
+	char	   *caller_name;
+
+	Cache	   *hypertable_cache;
+	Hypertable *ht;
+	Dimension  *time_dim;
+	Oid			time_dim_type = InvalidOid;
+
+	/*
+	 * contains the list of hypertables which need to be considred. this is a
+	 * list containing a single hypertable if PG_ARGISNULL(0) is false.
+	 * Otherwise, it will have the list of all hypertables in the system
+	 */
+	List	   *hypertables = NIL;
+	ListCell   *lc;
+	int			ht_index = 0;
+	long		num_chunks = 0;
+	MemoryContext oldcontext;
+	ChunkScanCtx **ctxs;
+	Chunk	  **chunks;
+	Chunk	  **current;
+
+
+	Oid			table_relid = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
+	Datum		older_than_datum = PG_GETARG_DATUM(1);
+	Datum		newer_than_datum = PG_GETARG_DATUM(2);
+
+	/*
+	 * get_fn_expr_argtype defaults to UNKNOWNOID if argument is NULL but
+	 * making it InvalidOid makes the logic simpler later
+	 */
+	Oid			older_than_type = PG_ARGISNULL(1) ? InvalidOid : get_fn_expr_argtype(fcinfo->flinfo, 1);
+	Oid			newer_than_type = PG_ARGISNULL(2) ? InvalidOid : get_fn_expr_argtype(fcinfo->flinfo, 2);
+
+	/*
+	 * chunks_return_srf is called even when it is not the first call but only
+	 * after doing some computation first
+	 */
+	if (!SRF_IS_FIRSTCALL())
+		return chunks_return_srf(fcinfo);
+
+	funcctx = SRF_FIRSTCALL_INIT();
```

Comment:
I think it might make sense to do firstcall stuff in its own function, like the non-firstcall case.